src/User32.Desktop/User32.cs

```diff
@@ -109,6 +128,12 @@ public static partial class User32
         [DllImport(nameof(User32))]
         public static extern int SendMessage(IntPtr hWnd, int wMsg, IntPtr wParam, IntPtr lParam);
 
+        [DllImport(nameof(User32))]
+        public static extern int SendMessage(IntPtr hWnd, int wMsg, IntPtr wParam, StringBuilder lParam);
+
+        [DllImport(nameof(User32))]
+        public static extern int SendMessage(IntPtr hWnd, int wMsg, uint wParam, int lParam);
```

Comment:
> As far as I understand pointers on C# is bad and Marshalling is the native way of handling with it. 

Ya, that's what I thought when starting this project as well. But I've since learned that C# pointers are often *easier* than marshaling and they are certainly much more flexible and more performant. For example, as soon as you add *any* marshaling attributes to a struct or its fields, that struct can no longer be pinned and sent to native code. So if you need to call a native method in a performance-critical loop, marshaling must get involved and copy the array of structs every time. Huge perf drain. 
Also, using `IntPtr` is a _real_ pain compared to just pinning an object and using pointers in C#. Much less room for error and memory leaks when using C# `fixed` keyword.
But you can achieve many of the benefits of marshaling but without sacrificing perf by using pointers.

> Given that and the fact that SendMessage changes the semantics of wParam and lParam according to the WIndows Message you are sending

No, I don't think it does. I think `SendMessage` is blind to the actual message and its parameters. It just relays data. It's the *receiver* that reads the message and uses that as a guide for how to interpret the parameters. Thus, there could literally be millions of parameter types accepted by all the receivers that exist as there is a range for custom messages that apps may define. 

> I think overloads are the best option to make it easier to the user/developer that is not so acquainted with native-managed marshalling and all the little quirks that it comes with.

That may be true, if there are sufficiently popular parameter types the warrant the overloads. We'd probably want some written guidance on what qualifies so that we don't have opinion battles in PRs over whether a particular overload is merited. That's why we have a rather large CONTRIB document: to avoid opinion battles during PRs. We just don't address this particular issue (yet). We should do that. The opinion battle is welcome during the deliberation of what should go into the guidance doc, and your opinion is of course welcome. We should have that over on [gitter.im](https://gitter.im/AArnott/pinvoke)

> The IntPtr marshalling helps us for pointers but it doesn't make sense to box/unbox it just for the sake of using IntPtr, at least for some of the Windows Messages that uses only values and not pointers.

`IntPtr` isn't a ref type, so converting integers to/from it isn't 'boxing'. I don't mean to be nit-picky, but the argument against boxing usually is merited by needing to avoid the GC pressure from allocating an object, and there is no such pressure in this case. 
The convenience of `int` over `IntPtr` is undoubted, but the inaccuracy of `int` being a 32-bit value whereas on Windows the value can vary in size is what forces us to use `IntPtr` to guarantee the accuracy of the method and that it works properly on any architecture. @vbfox was amazing in his review and so _perhaps_ it would work with just `int` out of luck. But if we were going to add an `int` overload, I think it would first need to start building this assembly for x86/x64 specifically and then use `int` and `long` for each using `#if`. But losing our "Any CPU" target would undoubtedly be a pain and cause a trickle-down effect to all folks who reference it, which IMO would outweigh the gain.

What we *can* do for a case like this though is add a helper overload. The helper can take `int` and convert it to `IntPtr` before forwarding to the P/Invoke method. I have no objection to that route -- after the guidance doc is written that can spell out which overloads will merit inclusion to avoid an overload explosion.